Add runtime changes to the 2201.6.0 release note (#6930)
ballerina-platform/ballerina-dev-website: downloads/swan-lake-release-notes/swan-lake-2201.6.0/RELEASE_NOTE.md

If you have not installed Ballerina, download the [installers](/downloads/#swanlake) to install.

## Backward-incompatible changes

> - Fixed a bug with the XML parser error messages that showed dependency information. The error message prefix "failed to create xml" is now changed to "failed to parse xml", to have a single prefix for all the XML parsing error messages.

Reviewer: ```suggestion
- Fixed a bug with the XML parser error messages that showed dependency information. The error message prefix `failed to create XML` is now changed to `failed to parse XML` to have a single prefix for all the XML-parsing error messages.
```

Author: Fixed with ddebb40fedf58faa740d98e4e6f5ac42a7a77a37.

>     For example, consider the following `invalid_xml.txt` file contents.

Reviewer: ```suggestion
    For example, consider the content below of the `invalid_xml.txt` file.
```

    ```
    <!-- comments cannot have -- in it -->
    ```

    When the following Ballerina code is run,

    ```ballerina
    import ballerina/io;

    public function main() returns error? {
        xml readResult = check io:fileReadXml("invalid_xml.txt");
    }
    ```

>     it gives

Reviewer: ```suggestion
it gives the error below.
```

Author: Fixed with ddebb40fedf58faa740d98e4e6f5ac42a7a77a37.

    ```
    error: failed to parse xml: String '--' not allowed in comment (missing '>'?)
     at [row,col {unknown-source}]: [1,4]
    ```

> - The `Environment`, `Future` and `Runtime` classes in the `io.ballerina.runtime.api` package are made abstract classes. Creating an instance of those classes is not correct.

Reviewer: ```suggestion
- The `Environment`, `Future`, and `Runtime` classes in the `io.ballerina.runtime.api` package are abstract classes. Creating an instance of those classes is not correct.
```

> - The `Environment`, `Future`, and `Runtime` classes in the `io.ballerina.runtime.api` package are abstract classes. Creating an instance of those classes is not correct.

Reviewer 2: ```suggestion
- The `Environment`, `Future`, and `Runtime` classes in the `io.ballerina.runtime.api` package are refactored to abstract classes. Creating an instance of those classes is not correct.
```

> - The `Environment`, `Future`, and `Runtime` classes in the `io.ballerina.runtime.api` package are refactored to abstract classes. Creating an instance of those classes is not correct.

Reviewer 3: ``` suggestion
- The `Environment`, `Future`, and `Runtime` classes in the `io.ballerina.runtime.api` package are refactored to abstract classes. Creating an instance of those classes is incorrect.
```

> - A typedesc value returned by evaluating the `typeof` expression on a result of the `value:cloneWithType` and `value:fromJsonWithType` functions is changed to give the correct type-reference type.

Reviewer: ```suggestion
- A typedesc value (returned by evaluating the `typeof` expression on a result of the `value:cloneWithType` and `value:fromJsonWithType` functions) is changed to give the correct type-reference type.
```

### New features

#### New Runtime Java APIs

> Introduced `getInitMethod()` API in the `io.ballerina.runtime.api.types.ObjectType` class to get the method type of the initializer method of Ballerina objects.

Reviewer: ```suggestion
Introduced the `getInitMethod()` API in the `io.ballerina.runtime.api.types.ObjectType` class to get the method type of the initializer method of Ballerina objects.
```

> Introduced the `getInitMethod()` API in the `io.ballerina.runtime.api.types.ObjectType` class to get the method type of the initializer method of Ballerina objects.

Reviewer 3: Shall we mention the method definition in java?

Author: It is not clear to me what you ask. Can you please suggest the change?

Reviewer 3: sure
``` suggestion
Introduced the `getInitMethod()` API in the `io.ballerina.runtime.api.types.ObjectType` class to get the method type of the initializer method of Ballerina objects.
```java
    MethodType getInitMethod();
```

```

Author: Thanks! Fixed with 62048b5164f393ba3319993cca4c793084000709.

### Improvements

#### Improvements in Runtime Java APIs

> The following APIs in the `io.ballerina.runtime.api` package are deprecated and marked for removal in a future release.

Author: Will do

> The following APIs in the `io.ballerina.runtime.api` package are deprecated and marked for removal in a future release.

Author: Btw I see the output is displayed as 
<img width="1044" alt="image" src="https://github.com/ballerina-platform/ballerina-dev-website/assets/33286289/12b73f35-4be2-4e92-8974-e9d9d7d730da">
May I remove the common `io.ballerina.runtime.api` part and keep only the remaining part in the Java class column?
Eg: `creators.ValueCreator`

> The following APIs in the `io.ballerina.runtime.api` package are deprecated and marked for removal in a future release.

Author: Resolving the comment since this is a problem with the Ballerina website formatting.

> The following APIs in the `io.ballerina.runtime.api` package are deprecated and marked for removal in a future release.

Reviewer 3: +1

> The following APIs in the `io.ballerina.runtime.api` package are deprecated and marked for removal in a future release.

Author: @sm1990 I changed the font size by converting the the table to html.
